### DISCUSSION TITLE
How to set optimal height for rows ?

### DISCUSSION THREAD

**Discussion by holicc** [qax-os/excelize#1327] [Q&A]
when cell's content exceed the width and height , how to set optimal property for the row ? 

or can i tell is the content  exceed cell's width or height, so i can change the cell's width or height to fit the content.

![image](https://user-images.githubusercontent.com/19146591/186062513-db073b78-544b-4751-b656-4c7713274ca1.png)

![image](https://user-images.githubusercontent.com/19146591/186063004-ce2c3d97-a3e1-48a2-9e23-41adfe2b78c0.png)

![image](https://user-images.githubusercontent.com/19146591/186063040-595c1c4a-deae-4677-b177-ff267eddf4ee.png)

**Comment by xuri** [comment]
Thanks for your feedback. This problem similar with the issues #92 and #204. This library not support this feature currently, and you can calculate the approximate row height according to the total characters in a cell, and then set the row's height or column width.

**Reply by holicc** [reply]
oh, it seems many people need this feature. i'd like to make a pr.

**Reply by xuri** [reply]
Contributions are welcome! As I said in issue #92, to auto-fit columns width and rows height, we need to calculate the column width based on the calculated value of the column (so on the result of any formulae, we need to implement the formula calculation engine first), and any additional characters added by format masks such as thousand separators (number format support), there a lot of work we need to do. That's an interesting job and I will consider adding support for this feature later. I'll certainly accept that patch if somebody did that.

**Reply by holicc** [reply]
maybe we don't need `formula calculation engine ` , just let user define the `width` or `height` function.
```go
func (f *File) AutoFitHeight(sheet string, row int, width func(string) float64) error
```

**Reply by xuri** [reply]
Sorry, I don't suggest adding this feature, that's not a real autofit. Currently, users can be using the `GetCellValue` function to get cell characters, calculate the column width or row height by themselves, and change cell size by the existing function `SetColWidth` or `SetRowHeigh` function. But you know, that unable to calculate the exact width and height of the cell just by characters, it also depends on the number formatting styles. So, I think this feature will be added after dependencies implemented.
